mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -318,6 +327,14 @@ private[ml] object DefaultParamsWriter {
     val metadataJson: String = compact(render(metadata))
     metadataJson
   }
+
+  def saveInitialModel[T <: HasInitialModel[_ <: MLWritable]](instance: T, path: String): Unit = {
+    if (instance.isDefined(instance.getParam("initialModel"))) {
```

Comment:
Since we specify the type parameters, you can access the param directly `instance.isDefined(instance.initialModel)`